src/app/modules/core/services/map.services.ts

```diff
@@ -1,87 +1,115 @@
   public CurrentZoomLevel;
-  public config;
-  public esriurl;
+  public CurrentLayer: String;
 
-  constructor(public http: HttpClient) {
+  constructor(http: HttpClient) {
 
-    this.options = {
-      layers: [
-        L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', { })
-      ],
+    this.Options = {
       zoom: 10,
       center: L.latLng(46.95, -122)
     };
 
-    this.CurrentZoomLevel = this.options.zoom;
-    // this.ActiveLayers.push(this.layersControl['baseLayers']['National Geographic']);
-    // this.ActiveLayers.push(this.layersControl['overlays']['State Cities']);
-    // this.ActiveLayers.push(this.layersControl['overlays']['Big Circle']);
-    // this.ActiveLayers.push(this.layersControl['overlays']['Big Square']);    
-  }
-
-
-  ngOnInit() {
-    this.http.get("../../../.././assets/config.json").subscribe(data => {
-      this.config = data;
+    http.get("../../../.././assets/config.json").subscribe(data => {
+      //load baselayers
+      var conf: any = data;
+      conf.mapLayers.baseLayers.forEach(ml => {
+        if (ml.visible) {
+          this.CurrentLayer = ml.name;
+        } else { }
+        ml.layer = this.loadLayer(ml);
+        if (ml.layer != null)
+          this._layersControl.baseLayers.push(ml);
+      });
+      conf.mapLayers.overlayLayers.forEach(ml => {
+        ml.layer = this.loadLayer(ml);
+        if (ml.layer != null)
+          this._layersControl.overlays.push(ml);
+      });
+      this.LayersControl.next(this._layersControl);
     });
 
-    this.layersControl = {
-      baseLayers: {
-        'National Geographic': esri.basemapLayer('NationalGeographic'),
-        'Open Street Map': L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', { maxZoom: 18, attribution: '...' }),
-        'Streets': esri.basemapLayer('Streets'),
-        'Topographic': esri.basemapLayer('Topographic')
-      },
-      //You can add any kind of Leaflet layer you want to the overlays map. This includes markers, shapes, geojson, custom layers from other libraries, etc.
-      overlays: {
-        'State Cities': this.addFeatureLayer(this.config.EsriFeatureLayer),
-        'Big Circle': L.circle([46.95, -122], { radius: 5000 }),
-        'Big Square': L.polygon([[46.8, -121.55], [46.9, -121.55], [46.9, -121.7], [46.8, -121.7]])
-      }
+    this.CurrentZoomLevel = this.Options.zoom;
+  }
+
+  public AddLayer(point: any) {
+    //this is just and example of how to add layers
+    var newlayer = {
+      name: 'Big Circle',
+      layer: L.circle(point, { radius: 5000 }),
+      visible: true
     };
+    var ml = this._layersControl.overlays.find((l: any) => (l.name === newlayer.name));
+    if (ml != null) ml.layer = newlayer.layer;
+    else this._layersControl.overlays.push(newlayer);
+
+    //Notify subscribers
+    this.LayersControl.next(this._layersControl);
   }
 
+  public ToggleLayerVisibility(layername: string, kind: string) {
```

Comment:
Similar to the map compent comments, lets split the baselayer and overlays into 2 methods
ToggleLayerVisibility() - for overlay layers
SwitchBaseMaps() - for basemap layers. 

this should give us a simpler, more concise way for dealing with the 2 types of layers